go.mod

```diff
@@ -26,17 +26,19 @@ require (
 	k8s.io/client-go v0.32.4
 	k8s.io/klog/v2 v2.130.1
 	k8s.io/utils v0.0.0-20241104100929-3ea5e8cea738
+	maistra.io/api v0.0.0-20240319144440-ffa91c765143
```

Comment:
do we actually need  this dependency ? it seems that we can reuse the same template file we have and conditionally include it.

Reply:
reverted back to use template with condition checks in `initialize`